
enum e_gsgm_proto_version{
	GSGM_VERSION = 1;
};

//gm发给gs的 有response和notify
enum e_msg_gm_gs{
	e_msg_gm_gs_min = 0 ;		//必须为0  从0开始 方便为了funcmap数组存储
	e_msg_gm_gs_marquee_req = 1;	//跑马灯请求
	e_msg_gm_gs_add_item_req = 2;	//给玩家添加道具的请求
	e_msg_gm_gs_mail_req = 3;
	e_msg_gm_gs_arena_mail_req = 4;	//竞技场结算奖励邮件(每天9点发送)
	e_msg_gm_gs_max = 5;		
};

//gs发给gm的 有response和notify
enum e_msg_gs_gm{
	e_msg_gs_gm_min = 0 ;		//必须为0  从0开始 方便为了funcmap数组存储
	e_msg_gs_gm_marquee_res = 1;	//
	e_msg_gs_gm_add_item_res = 2;	//
	e_msg_gs_gm_mail_res = 3;
	e_msg_gs_gm_arena_mail_res = 4;
	e_msg_gs_gm_max = 5;
};

enum e_gsgm_errorcode{
	e_gsgm_errorcode_min	= 0 ;			//因为初始化就为0 所以这里作为初始化
	e_gsgm_errorcode_success = 1 ;			//操作成功
	e_gsgm_errorcode_unkown_error = 2 ;		//未知错误
	e_gsgm_errorcode_user_not_online = 3;		//不在线
	e_gsgm_errorcode_max = 4;
};

message game_element{
	required uint32 type = 1;
	required uint32 id = 2;
	required uint32 num = 3; //这里的数字是增量
};

//<<< common -------------

message gm_gs_add_item_req{
	required uint32 user_id = 1;
	required uint32 item_id = 2;
	required uint32 num = 3;
};

message gs_gm_add_item_res{
	required int32 error_code = 1;
};

message gm_gs_marquee_req{
	enum MARQUEE_TYPE{
		//MARQUEE_MIN = 0;
		//MARQUEE_INVOKE = 1;		//祭坛抽取
		//MARQUEE_HERO_EVOLUTION = 2;	//英雄突破
		//MARQUEE_INVOKE_HERO = 3;	//召唤英雄
		//MARQUEE_HERO_STAR_UP = 4;	//英雄升星
		//MARQUEE_ARENA_CHALLENGE = 5;	//竞技场挑战
		MARQUEE_WORD = 6;		//自定义(自定义一句话，客户端直接显示)
		//MARQUEE_MAX = 7;
	};
	required MARQUEE_TYPE type = 1;		//跑马灯类型
	//required uint32 times = 2;		//滚动次数 次数来自配置！
	required string content = 2;		//内容(以逗号隔开)
}

message gs_gm_marquee_res{
	required int32 error_code = 1;
};

message gm_gs_mail_req{
	enum MAIL_TYPE
	{
		//MAIL_TYPE_RECHARGE_SUCCESS = 1;		//充值成功
		//MAIL_TYPE_VIP_LEVEL_UP = 2;		//VIP等级提升
		//MAIL_TYPE_ARENA_RANK_REWARD =  3;	//竞技场排名奖励
		//MAIL_TYPE_ARENA_TOP_RANK = 4;		//竞技场头名奖励	
		//MAIL_TYPE_OVERLAP_ITEMS = 5;		//物品到达堆叠上限
		MAIL_TYPE_SYS_NOTICE = 6;		//系统公告
		//MAIL_TYPE_PRE_MONTH_ADD_SIGN = 7;	//上月补签物品
		//MAIL_TYPE_MAIL_GUIDE = 8;		//邮箱引导奖励
	};
	required MAIL_TYPE type = 1;			//邮件类型
	repeated uint32 users = 2;			//发送玩家
	required string	content = 3;			//具体内容
	repeated game_element elements = 4;		//相关物品，金币等	
};

message gs_gm_mail_res{
	required int32 error_code = 1;
};

message gm_gs_arena_mail_req{
}

message gs_gm_arena_mail_res{
	required int32 error_code = 1;
}